### ISSUE TITLE
Style/MultilineWhenThen breaks case statements with {}

### ISSUE THREAD

**Issue by tas50** [rubocop-hq/rubocop#7616]
## Expected behavior

The Style/MultilineWhenThen cop should remove then statements from all mulitline case statements w/o creating invalid ruby code.

## Actual behavior

If RuboCop encounters a case statement that uses then and {}s it will just remove the then statements, which results in invalid ruby:

## Steps to reproduce the problem

Run `rubocop --only Style/MultilineWhenThen` on the following code:

```ruby
      case version.to_s # to_s to make sure someone didn't pass us an int
      when '9.0' then {
        'ActiveDirectory' => 'Veeam Explorer for Microsoft Active Directory',
        'SQL' => 'Veeam Explorer for Microsoft SQL Server',
        'Exchange' => 'Veeam Explorer for Microsoft Exchange',
        'SharePoint' => 'Veeam Explorer for Microsoft SharePoint',
        'Oracle' => 'Veeam Explorer for Oracle'
      }
     end
```

This will autocorrect to:

```ruby
      case version.to_s # to_s to make sure someone didn't pass us an int
      when '9.0' {
        'ActiveDirectory' => 'Veeam Explorer for Microsoft Active Directory',
        'SQL' => 'Veeam Explorer for Microsoft SQL Server',
        'Exchange' => 'Veeam Explorer for Microsoft Exchange',
        'SharePoint' => 'Veeam Explorer for Microsoft SharePoint',
        'Oracle' => 'Veeam Explorer for Oracle'
      }
     end
```

## RuboCop version

```
0.78.0
```
